docs/log-levels.md

```diff
@@ -14,28 +14,29 @@ SwiftLog defines the following 7 log levels via the [`Logger.Level` enum](https:
 
 This is in contrast with `debug` which some users _may_ choose to run enabled on their production systems.
 
 > Debug level logging should be not "too" noisy. Developers should assume some production deployments may need to (or want to) run with debug level logging enabled. 
 >
 > Debug level logging should not completely undermine the performance of a production system.
 
-As such, `debug` logging should not be overly noisy. It should provide a high value understanding of what is going on in the library for end users without diving deep into its internals. This is what `trace` is intended for.
+As such, `debug` logging should provide a high value understanding of what is going on in the library for end users, using domain relevant language.  Logging at `debug` level should not be overly noisy or dive deep into internals; this is what `trace` is intended for.
 
-Use `warning` level sparingly. Whenever possible, try to rather return or throw `Error` to end users that are descriptive enough so they can inspect, log them and figure out the issue. Potentially they may then enable debug logging to find out more about the issue.
+Use `warning` level sparingly. Whenever possible, try to rather return or throw `Error` to end users that are descriptive enough so they can inspect, log them and figure out the issue. Potentially, they may then enable debug logging to find out more about the issue.
 
 It is okay to log a `warning` "once", for example on system startup. This may include some one off "more secure configuration is available, try upgrading to it!" log statement upon a server's startup. You may also log warnings from background processes, which otherwise have no other means of informing the end user about some issue.
 
-Logging on `error` level is similar to warnings: prefer to avoid doing so whenever possible. Instead, report errors via your library's API. For example, it is _not_ a good idea to log "connection failed" from an HTTP client. Perhaps the end-user intended to make this request to a known offline server to _confirm_ it is offline? From their perspective this connection error is not a "real" error, it is just what they expected -- as such the HTTP client should return or throw such an error, but _not_ log it.
+Logging on `error` level is similar to warnings: prefer to avoid doing so whenever possible. Instead, report errors via your library's API. For example, it is _not_ a good idea to log "connection failed" from an HTTP client. Perhaps the end-user intended to make this request to a known offline server to _confirm_ it is offline? From their perspective, this connection error is not a "real" error, it is just what they expected -- as such the HTTP client should return or throw such an error, but _not_ log it.
```

Comment:
True, and there's also another problem with errors in that they can require context that is lacking.  I'm not sure where to put this, so I'll just describe it and hope it's useful.  This isn't an argument against error log levels per ce but more about how to handle operations when work is offloaded "somewhere else".

Akka HTTP has a parser that runs asynchronously on HTTP POST input - the network deals with the chunks and the parser deals with the input.  However, there are situations where the incoming POST dies and terminates the connection.  This then kills the connection to the parser, but the parser doesn't know what connection and what POST request was responsible, or even what the input was at all.  As such, it logs an error because it has no feedback channel to know what happened and why.

So even though there's an error, it's not useful in itself -- there's no context, and no way for the network processor to signal to the parser than it should give up and die quietly.

Reply:
Yeah, I'll see if any useful wording about it that i could add